Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs

```diff
@@ -0,0 +1,160 @@
+        }
+    }
+
+    public TMToastContents()
+    {
+    }
+
+    private void CloseButton_Clicked(object sender, EventArgs e)
+    {
+         popupNavigation.RemovePageAsync(this, true);
+       
+    }
+    public void Close()
+    {
+        Task.Run(async () => {
+            await Task.Delay(5000);
+            await popupNavigation.RemovePageAsync(this, true);
+         });
+    }
+
+    private void PopupData(ImageSource leftIcon, string message, string rightIconText)
+    {
+       
+        LeftIconSource = leftIcon;
+
+        RightIconText = rightIconText;
+        TMButton rightIcon = new TMButton();
+        rightIcon.Title = RightIconText;
+        rightIcon.TextColor = TextColor;
+        if (string.IsNullOrEmpty(RightIconText)) {
+            if(ToastTheme.Equals((Color)BaseComponent.colorsDictionary()["ToastBlue"]))
+            {
+                rightIcon.IconSource = ImageSource.FromResource("Trimble.Modus.Components.Images.blue_close_icon.png");
+            }
+            else if (ToastTheme.Equals((Color)BaseComponent.colorsDictionary()["ToastBlack"]))
+            {
+                rightIcon.IconSource = ImageSource.FromResource("Trimble.Modus.Components.Images.white_close_icon.png");
+            }
+            else
+            {
+                rightIcon.IconSource = ImageSource.FromResource("Trimble.Modus.Components.Images.black_close_icon.png");
+            }
+        }
+        rightIcon.VerticalOptions = LayoutOptions.Center;
+        rightIcon.HorizontalOptions = LayoutOptions.End;
+        rightIcon.BackgroundColor = this.BackgroundColor;
+
+        rightIcon.Size = Enums.Size.XSmall;
+        rightIcon._iconWidth = 16;
+        rightIcon._iconHeight = 16;
+
+        rightIcon.BorderColor = Colors.Transparent;
+        rightIcon.Clicked += CloseButton_Clicked;
+        contentLayout.Children.Add(rightIcon);
+
+        var idiom = Device.Idiom;
+
+        setWidth(rightIcon,idiom);
+     
+        Message = GetWrappedLabelText(message,idiom);
+    }
+
+    private void setWidth(TMButton rightIcon,TargetIdiom idiom)
+    {
+        double minimumTabletWidth = 480; 
+        double maximumTabletWidthPercentage = 0.7;
+        double deviceWidth = DeviceDisplay.MainDisplayInfo.Width;
+    
+        if (idiom == TargetIdiom.Phone)
+        {
+            toastLayout.Padding = new Thickness(16,0,16,10);
+            rightIcon.Size = Enums.Size.XSmall;
+        }
+        else if (idiom == TargetIdiom.Tablet)
+        {
+            toastLayout.Padding = new Thickness(0, 0, 0, 10);
+            toastLayout.MinimumWidthRequest = minimumTabletWidth;
+            toastLayout.MaximumWidthRequest = deviceWidth * maximumTabletWidthPercentage;
+            toastLayout.HorizontalOptions = LayoutOptions.CenterAndExpand;
+            rightIcon.Size = Enums.Size.Large;
+        }
+
+    }
+    private string GetWrappedLabelText(string text, TargetIdiom idiom)
```

Comment:
Huge red flag. We shouldn't need this. If we set LineBreakMode, it toast should automatically adds the ellipsis. Can you create a improvement task in the github and attach the link here? We can take it in the next sprint

Reply:
Sure

Reply:
the issue for removing the separate method : https://github.com/trimble-oss/modus-mobile-maui-components/issues/73